src/timeman.cpp

```diff
@@ -65,26 +68,30 @@ void TimeManagement::init(Search::LimitsType& limits,
     // to nodes, and use resulting values in time management formulas.
     // WARNING: to avoid time losses, the given npmsec (nodes per millisecond)
     // must be much lower than the real engine speed.
-    if (npmsec)
+    if (useNodesTime)
     {
-        useNodesTime = true;
-
-        if (!availableNodes)                            // Only once at game start
+        if (availableNodes == -1)                       // Only once at game start
             availableNodes = npmsec * limits.time[us];  // Time is in msec
 
         // Convert from milliseconds to nodes
         limits.time[us] = TimePoint(availableNodes);
         limits.inc[us] *= npmsec;
         limits.npmsec = npmsec;
+        moveOverhead *= npmsec;
     }
 
+    // These are used to convert numbers when using nodestime.
+    const int64_t M = useNodesTime ? npmsec : 1;
+    const auto    C = [&](int64_t c) -> int64_t { return c * M; };
```

Comment:
can we have a descriptive name for these functions, I don't like so much single letter function names.

Reply:
also just take M as a parameter and make this constexpr static

Reply:
or well mh i guess can leave it that way now for nicer syntax later,  but yeah overall this could use some better explanation

Reply:
I can rewrite function names, but I didn't really like using lambda functions. The only purpose of them is to emphasize which part is changed. Maybe @Disservin can help with the code design/structure ...